### PR TITLE
Table lookup matching definitions and then ask user

### LINKED ISSUE #581: TableManager Overly Strict Filename to Definitions Matching
Tablemanager currently requires table files to exactly match the filename with their definitions.  It should support looser automatic matching, and allow for manual definition selection if ambigious.

### PR DESCRIPTION
**Pull request by jmthomas**
closes #581

### CODE REVIEW

**Review by ryanmelt — CHANGES_REQUESTED** [review]
So what happens now if the definition can't be found?

File: openc3-cosmos-cmd-tlm-api/app/controllers/tables_controller.rb [hunk 1]
```diff
@@ -66,20 +57,16 @@ def report
     return unless authorization('system')
     begin
       file = Table.report(params[:scope], params[:binary], params[:definition], params[:table])
-      if file
-        results = { 'filename' => file.filename, 'contents' => file.contents }
-        render json: results
-      else
-        head :not_found
-      end
-    rescue Exception => e
```
[the remark this answers is not in the document]
Comment (jmthomas) [reply openc3-cosmos-cmd-tlm-api/app/controllers/tables_controller.rb:75]: The default Rails handler sends the exception. I'm not doing anything special and it's already 500. I'll verify but I think I tried that.

File: openc3/lib/openc3/tools/table_manager/table_manager_core.rb [hunk 2]
```diff
@@ -189,7 +191,7 @@ def self.build_json(binary, definition_filename)
           end
         end
       end
-      json.as_json(:allow_nan => true).to_json(:allow_nan => true)
```
[the remark this answers is not in the document]
Comment (jmthomas) [reply openc3/lib/openc3/tools/table_manager/table_manager_core.rb:192]: I moved both to the caller but I can leave that half as long as that's still returning a hash.

**Comment by jmthomas** [comment]
> So what happens now if the definition can't be found?

I have a playwright test for this. It pops up a green warning that says it can't be found along with the File Open dialog for you to select the config file.

**Review by ryanmelt — APPROVED** [review]

**Comment by jmthomas** [comment]
Default error handler:
Screenshot 2023-04-23 at 3 06 53 PM

**Pull request merged**